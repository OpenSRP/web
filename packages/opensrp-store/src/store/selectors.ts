import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';
import { Dictionary } from '@onaio/utils';
import { createSelector } from 'reselect';

/** get API Token from the Redux store
 *
 * @param {any} state - the redux store
 */
interface StoreState {
  session: {
    [key: string]: Dictionary;
  };
}

interface APIFilters {
  accessToken?: boolean;
  apiToken?: boolean;
  providerState?: boolean;
}

/** Fetch access token from store if access_token filter val is true
 *
 * @param _ - redux store
 * @param props - api filters object
 * @returns {boolean}
 */

export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.accessToken || null;

/** Fetch api token from store if api_token filter val is true
 *
 * @param _ redux store
 * @param props api filters object
 * @returns {boolean}
 */

export const fetchApiToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.apiToken || null;

/** Fetch oauth provider state from store if providerState filter val is true
 *
 * @param _ redux store
 * @param props api filters object
 * @returns {boolean}
 */

export const fetchOauthProviderState = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.providerState || null;

/** Gets extra data object from store
 *
 * @param state - redux store
 */

export const getExtraData = (state: Partial<Store>): Dictionary =>
  (state as StoreState)[sessionReducerName].extraData;

/**
 * @param {any} state - the redux store
 * @returns {Object} - the states
 */

export const makeAPIStateSelector = () =>
  createSelector(
    [fetchAccessToken, fetchApiToken, fetchOauthProviderState, getExtraData],
    (accessToken, apiToken, oauthState, extraData) => {
      if (accessToken) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.access_token) || null;
      } else if (apiToken) {
        return extraData.api_token || null;
      } else if (oauthState) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.state) || null;
      }
    }
  );
