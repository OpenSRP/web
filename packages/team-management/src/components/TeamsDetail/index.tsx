import * as React from 'react';
import { CloseOutlined } from '@ant-design/icons';
import { Button } from 'antd';
import { Organization } from '../../ducks/organizations';

export interface TeamsDetailProps extends Organization {
  onClose?: Function;
}

const TeamsDetail = (props: TeamsDetailProps) => {
  const { name, active, identifier } = props;
  return (
    <div className="p-4 bg-white">
      <Button
        shape="circle"
        onClick={() => (props.onClose ? props.onClose() : '')}
        className="float-right"
        type="text"
        icon={<CloseOutlined />}
      />
      <div className="mb-4 small mt-4">
        <p className="mb-0 font-weight-bold">Team name</p>
        <p className="mb-0">{name}</p>
      </div>
      <div className="mb-4 small">
        <p className="mb-0 font-weight-bold">Status</p>
        <p className="mb-0">{`${active}`}</p>
      </div>
      <div className="mb-4 small">
        <p className="mb-0 font-weight-bold">Identifier</p>
        <p className="mb-0">{`${identifier}`}</p>
      </div>
      {/* <div className="mb-4 small">
        <p className="mb-0 font-weight-bold">Created</p>
        <p className="mb-0">{`${date}`}</p>
      </div>
      <div className="mb-4 small">
        <p className="mb-0 font-weight-bold">Last updated</p>
        <p className="mb-0">{`${active}`}</p>
      </div>
      <div className="mb-4 small">
        <p className="mb-0 font-weight-bold">Team members</p>
        <p className="mb-0">{`${active}`}</p>
      </div> */}
    </div>
  );
};

export default TeamsDetail;
