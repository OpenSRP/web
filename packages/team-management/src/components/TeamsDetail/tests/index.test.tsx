import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import flushPromises from 'flush-promises';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { store } from '@opensrp/store';
import TeamsDetail, { Props } from '..';

describe('containers/pages/Home', () => {
  const props: Props = {
    active: true,
    id: 1,
    identifier: 'fcc19470-d599-11e9-bb65-2a2ae2dbcce4',
    name: 'The Luang',
    type: {
      coding: [
        {
          code: 'team',
          display: 'Team',
          system: 'http://terminology.hl7.org/CodeSystem/team-type',
        },
      ],
    },
  };

  it('renders without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
        <TeamsDetail {...props} />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('removes it self on close', async () => {
    const wrapper = mount(
      <Provider store={store}>
        <TeamsDetail {...props} onClose={() => wrapper.unmount()} />
      </Provider>
    );
    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    expect(wrapper.children()).toHaveLength(1);
    // wrapper.find('.close-btn').simulate('click');
    // expect(wrapper).toHaveLength(0);
  });
  // it('doesnt close if onClose prop is not set', () => {
  //   const wrapper = mount(
  //     <Provider store={store}>
  //       <TeamsDetail {...props} />
  //     </Provider>
  //   );
  //   expect(wrapper.children()).toHaveLength(1);
  //   wrapper.find('button').simulate('click');
  //   expect(wrapper).toHaveLength(1);
  // });
});
