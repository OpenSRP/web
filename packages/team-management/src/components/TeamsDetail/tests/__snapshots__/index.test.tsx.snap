// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/Home renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <TeamsDetail
    active={true}
    id={1}
    identifier="fcc19470-d599-11e9-bb65-2a2ae2dbcce4"
    name="The Luang"
    type={
      Object {
        "coding": Array [
          Object {
            "code": "team",
            "display": "Team",
            "system": "http://terminology.hl7.org/CodeSystem/team-type",
          },
        ],
      }
    }
  >
    <Spin
      size="small"
      spinning={true}
      wrapperClassName=""
    >
      <div
        className="ant-spin ant-spin-sm ant-spin-spinning"
      >
        <span
          className="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
        </span>
      </div>
    </Spin>
  </TeamsDetail>
</Provider>
`;
